File: 0724-find-pivot-index/0724-find-pivot-index.rb
# @param {Integer[]} nums
# @return {Integer}
def pivot_index(nums)
    left_sum=0
    right_sum=nums.reduce(:+)
    i=0
    while i<nums.length
        right_sum-=nums[i]
        return i if left_sum==right_sum
        left_sum+=nums[i]
        i+=1
    end
    return -1
end

File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    let lsum=0, rsum=0;
    let iterator = 0;
    do{
        rsum+=nums[iterator]
        iterator++
    }while(iterator<nums.length);

iterator=0
    do{
        rsum-=nums[iterator]
        if(lsum==rsum)
        return iterator
        lsum+=nums[iterator]
        iterator++
    }while(iterator<nums.length)

    return -1;
};